app.py

```diff
@@ -161,7 +168,7 @@ def get_current_frame_range(self, engine):
 
         return (current_in, current_out)
 
-    def set_frame_range(self, engine, in_frame, out_frame):
+    def set_frame_range(self, engine, in_frame, out_frame, prev_in, prev_out):
```

Comment:
I would have try to not break the API by having new params as kwargs and handle the case where they are not defined (especially since they seems to be used in a text filed only ?)  But I guess you check all set_frame_range calls ?